core/services/ardupilot_manager/mavlink_proxy/test_all.py

```diff
@@ -106,29 +102,28 @@ def test_mavlink_router() -> None:
     assert mavlink_router.set_logdir(pathlib.Path(".")), "Local path as MAVLinkRouter log directory failed."
     assert re.search(r"\d+", str(mavlink_router.version())) is not None, "Version does not follow pattern."
 
-    endpoint_1 = Endpoint("udpout", "0.0.0.0", 14551)
-    endpoint_2 = Endpoint("tcpin", "0.0.0.0", 14552)
-    endpoint_3 = Endpoint("tcpout", "0.0.0.0", 14553)
-    assert mavlink_router.add_endpoint(endpoint_1), "Failed to add first endpoint"
-    assert mavlink_router.add_endpoint(endpoint_2), "Failed to add second endpoint"
-    assert mavlink_router.add_endpoint(endpoint_3), "Failed to add third endpoint"
-    assert mavlink_router.endpoints() == [
-        endpoint_1,
-        endpoint_2,
-        endpoint_3,
-    ], "Endpoint list does not match."
-    with pytest.raises(NotImplementedError):
-        mavlink_router.add_endpoint(Endpoint("udpin", "0.0.0.0", 14551))
-    with pytest.raises(NotImplementedError):
-        mavlink_router.add_endpoint(Endpoint("serial", "/dev/autopilot", 115200))
-
-    assert mavlink_router.start(Endpoint("udpin", "0.0.0.0", 14550)), "Failed to start MAVLinkRouter"
-    assert mavlink_router.is_running(), "MAVLinkRouter is not running after start."
-    assert mavlink_router.exit(), "MAVLinkRouter could not stop."
-    assert mavlink_router.start(Endpoint("serial", "/dev/autopilot", 115200)), "Failed to start MAVLinkRouter"
-    assert mavlink_router.is_running(), "MAVLinkRouter is not running after start."
-    assert mavlink_router.exit(), "MAVLinkRouter could not stop."
-    with pytest.raises(NotImplementedError):
-        mavlink_router.start(Endpoint("udpout", "0.0.0.0", 14550))
-    with pytest.raises(NotImplementedError):
-        mavlink_router.start(Endpoint("tcpout", "0.0.0.0", 14550))
+    allowed_output_types = [EndpointType.UDPClient, EndpointType.TCPServer, EndpointType.TCPClient]
+    allowed_master_types = [EndpointType.UDPServer, EndpointType.Serial, EndpointType.TCPServer]
+    run_common_routing_tests(mavlink_router, allowed_output_types, allowed_master_types)
+
+
+def run_common_routing_tests(
+    router: AbstractRouter, allowed_output_types: List[EndpointType], allowed_master_types: List[EndpointType]
+) -> None:
+    allowed_output_endpoints = [e for e in OUTPUT_ENDPOINTS if e.connection_type in allowed_output_types]
```

Comment:
It's clear to use the filter function, since this will make more clear for someone what are you doing.
```python
filter(lambda endpoint: endpoint.connection_type in allowed_output_types, OUTPUT_ENDPOINTS)
```
Also, be aware to avoid single letter variables

Reply:
> You can also merge [0d63e2a](https://github.com/bluerobotics/companion-docker/commit/0d63e2aea44c7c215713c0426a5e95a2b0bce0b9) with [601bd90](https://github.com/bluerobotics/companion-docker/commit/601bd90d544d98f7196692e696ee203cd7bbc44f)

They seen to be different, I think. The first one extends the test coverage, while the last just refactor the code, keeping the functionality. From a commiting point of view one wouldn't know if a problem raised from extending the tests or refactoring them.

Reply:
This just creates unnecessary commits, you can refactor the code first and add the extra endpoints next, not the other way around. But ok.

Reply:
> Also, be aware to avoid single letter variables

About this: I'm trying to use just in cases where it's really explicitly what the variable are and writing the entire name will end up breaking the 120 characters limit and end up with a two or even three-lines expression, which becomes ugly and less readable.

In your example, for example, it ends up like that:
```
allowed_output_endpoints = filter(
    lambda endpoint: endpoint.connection_type in allowed_output_types, OUTPUT_ENDPOINTS
)
```

Reply:
Just to ilustrate, this is the expression with the single letter variable:

```python
allowed_output_endpoints = filter(lambda e: e.connection_type in allowed_output_types, OUTPUT_ENDPOINTS)
```

What do you think it better (honest question)?

Reply:
The extra space is not a problem, there is plenty of space, the important thing in code is readability and maintainability, a single letter is not maintainable and clear. For the same reason the code is being worked from a for loop to a filter, since this shows what is the intended behavior in code.